vespa/application.py

```diff
@@ -802,8 +798,39 @@ def get_document_v1_path(
         return f"/document/v1/{namespace}/{schema}/docid/{id}"
 
 
+class CustomHTTPAdapter(HTTPAdapter):
+    def __init__(self, *args, **kwargs):
+        self.pool_maxsize = kwargs.pop("pool_maxsize")
+        self.pool_connections = kwargs.pop("pool_connections")
+        self.num_retries_429 = kwargs.pop("num_retries_429", 10)
+        super().__init__(*args, **kwargs)
+
+        retry_strategy = Retry(
+            total=3,
+            backoff_factor=1,
+            raise_on_status=False,
+            status_forcelist=[503],
+            allowed_methods=["POST", "GET", "DELETE", "PUT"],
+        )
+        self.retry_strategy = retry_strategy
+
+    def send(self, request, **kwargs) -> Response:
+        for attempt in range(self.num_retries_429):
+            response = super().send(request, **kwargs)
+            if response.status_code == 429:
+                wait_time = (
+                    2** attempt + random.uniform(0, 1)
```

Comment:
```suggestion
                    0.1 * (1.618** attempt) + random.uniform(0, 1)
```

Reply:
Had a commit, but got distracted from pushing it. 
Great that you saw it! 👍